### PR TITLE
115: form attachments update for Central v2025.1, set content-type

### LINKED ISSUE #115: Update form attachment upload endpoint for Central 2025.1
This PR https://github.com/getodk/central-backend/pull/1471 which was released in 2025.1 last week changed the form upload endpoint to return an `Attachment` JSON object rather than `{"success": True}`. Since the new response object doesn't contain a `success` key anymore, a KeyError will be raised. This means pyodk currently can't be used to upload forms with attachments, unless using the less convenient client.post etc methods.

There may have also been other recent changes which lead to the Content-Type problem raised in #114 - I don't think Central used to expect a Content-Type, or if it did there wasn't an error. In any case for this problem it seems like the form attachment upload function should do the same mimetype guessing as in submission attachment upload (rather than just doing so for CSV as in #114). It might also mean that https://github.com/getodk/central/issues/751 is (now) an issue for form attachments as well.

### PR DESCRIPTION
**Pull request by lindsay-stevens**
Closes #115

- the form attachment upload endpoint now returns a json object representing attributes of the file, rather than a generic status message `{"success": True}`
- set a content-type header for form attachments, in the same way as for submission attachments, since both seem to require it when Central uses S3 file storage
  - it seems webforms errors out if no content-type is set, but enketo is OK
- also in test_client.py change the print() calls to basic non-none assertions since staging now has quite a lot of data which means those print calls are very noisy.

#### What has been done to verify that this works as intended?

Added unit test to check the content-type change, ran integration tests to check generally. Opened forms created before the change (enketo OK, webforms errored) and after the change (enketo OK, webforms different probably unrelated error - details in #115).

#### Why is this the best possible solution? Were any other approaches considered?

This follows the attachment handling for submission attachments, which seems appropriate since it appears Central treats these attachment types in a similar way (in regards to S3 storage etc).

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

The API change should allow pyodk to upload forms with attachments with Central 2025.1 (or later). The content-type change should allow forms with attachments uploaded via pyodk to work with webforms.

I'm not aware of workarounds or fixes for file attachments (submissions or forms) that have inaccurate content-types saved for them (sending the same file again doesn't seem to overwrite).

#### Do we need any specific form for testing your changes? If so, please attach one.

The tests include the required files / data etc.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.

The form attachment upload method is not directly exposed by the pyodk API so it would not require a docs update.

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyodk tests` and `ruff check pyodk tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: pyodk/_endpoints/form_draft_attachments.py [hunk 1]
```diff
@@ -52,18 +64,34 @@ def upload(
...
+                form_id=fid,
+                fname=file_name,
+            ),
+            logger=log,
+            headers=headers,
+            data=file_stream(),
+        )
         data = response.json()
-        return data["success"]
+        return FormAttachment(**data)
```
Comment (lognaturel) [inline pyodk/_endpoints/form_draft_attachments.py:97]: What happens if someone uses this code with an old Central version that has the `success` response? This will crash, right?

Comment (lindsay-stevens) [reply pyodk/_endpoints/form_draft_attachments.py:97]: Yes it would raise an error, due to the expected keys for `FormAttachment` not being present in the older response object format `{"success": True}`. Although, if a user has an older Central version then they wouldn't need this fix. Currently the pyodk code doesn't really address backwards compatibility so there may be other issues with older Central versions - I only run the E2E tests against dev/staging.

Alternatively if you think the attachment details are not important then this function could ignore the response and return None, or continue returning bool if it gets anything at all, or bool based on the `FormAttachment.exists` property. Having it possibly return one of multiple types with something like, detect the Central version, and return `bool | FormAttachment` I think would make the API too complicated.

Comment (lognaturel) [reply pyodk/_endpoints/form_draft_attachments.py:97]: > Currently the pyodk code doesn't really address backwards compatibility so there may be other issues with older Central versions - I only run the E2E tests against dev/staging.

I think that’s a fine approach. It’s uncommon for us to make breaking changes like this!

> Having it possibly return one of multiple types with something like, detect the Central version, and return bool | FormAttachment I think would make the API too complicated.

Agreed. 
 
 > bool based on the FormAttachment.exists property

That’s sounding good to me! I think for pyodk usage the returned metadata is not likely to be generally useful. Hiding the breaking change from users seems more valuable.
